forte/mrdsrg-spin-integrated/dsrg_mrpt2.h

```diff
@@ -176,6 +179,259 @@ class DSRG_MRPT2 : public MASTER_DSRG {
     /// Unitary matrix to block diagonal Fock
     ambit::BlockedTensor U_;
 
+    // NOTICE These are essential variables and functions for computing DSRG-MRPT2 gradient.
+    // Some variables may be redundant thus need further elimination
+    /// Set ambit tensor labels
+    void set_ambit_space();
+    /**
+     * Initialize tensors.
+     *
+     * Density tensors, one- and two-electron integrals, Fock matrix, DSRG and CI tensors.
+     */
+    void set_tensor();
+    /**
+     * Initializing the two-body density Gamma2_.
+     *
+     * NOTICE: this function shall be revoked in the future.
+     */
+    void set_density();
+    /**
+     * Initializing the Fock F.
+     *
+     * NOTICE: this function shall be revoked in the future.
+     */
+    void set_active_fock();
+    /**
+     * Initializing the one-electron integral H.
+     *
+     * NOTICE: this function shall be revoked in the future.
+     */
+    void set_h();
+    /**
+     * Initializing the ERIs V.
+     *
+     * NOTICE: this function shall be revoked in the future.
+     */
+    void set_v();
+    /// Set CI-relevant integrals
+    void set_ci_ints();
+    /**
+     * Initialize global variables.
+     *
+     * MO indices list, dimension of different MOs, CI-related variables etc.
+     */
+    void set_global_variables();
+    size_t nmo;
```

Comment:
I am surprised many of these are not already defined in the base class. Maybe @lcyyork can double check and eliminate if needed?

Reply:
No, they are not set because they can be obtained by saying something like `core_mos.size()` or from `mo_space_info_`.

BTW, the naming convention is violated. We have an `_` after the variable.